### PR TITLE
Remove scope from marketplace registration definitions operations

### PR DESCRIPTION
**Pull request by skayani**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [ ] new API version 
      - [ ] update existing version for new feature 
      - [x] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [x] SDK of .NET (need service team to ensure code readiness)
      - [x] SDK of Python
      - [x] SDK of Java
      - [ ] SDK of Js
      - [x] SDK of Go
      - [x] PowerShell
      - [x] CLI
      - [x] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [ ] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [ ] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by skayani** [comment]
The 'scope' part of the URI path that was removed is optional for marketplace registration definition operations. Initially I tried replacing the 'scope' parameter with a new 'optionalScope' parameter that was required=false but this lead to semantic validation errors so I believe this is the right approach. Please correct me if I'm wrong.

File: specification/managedservices/resource-manager/Microsoft.ManagedServices/preview/2020-02-01-preview/managedservices.json [hunk 1]
```diff
@@ -380,17 +380,14 @@
         }
       }
     },
-    "/{scope}/providers/Microsoft.ManagedServices/marketplaceRegistrationDefinitions": {
```
Comment (dramuy) [inline specification/managedservices/resource-manager/Microsoft.ManagedServices/preview/2020-02-01-preview/managedservices.json:383]: suggest to have {scopeOptional} as this API supports tenant and subscription level as well.

Comment (dramuy) [reply specification/managedservices/resource-manager/Microsoft.ManagedServices/preview/2020-02-01-preview/managedservices.json:383]: agree with removing; as I see for operations API also we have removed it.

**Comment by ArcturusZhang** [comment]
Hi @skayani we need to go through the breaking change review because this PR is introducing breaking changes. You could find the guidance in a comment by @openapi-workflow-bot

**Comment by skayani** [comment]
> Hi @skayani we need to go through the breaking change review because this PR is introducing breaking changes. You could find the guidance in a comment by @openapi-workflow-bot

I've gone ahead and done that, thanks

**Comment by skayani** [comment]
Hi @ArcturusZhang , I wanted to ask this question directly to you as in case I don't get a response from the Swagger tools team alias.  Here it is:

I tried looking for this in the docs but couldn't find a similar scenario. For our service we have these Azure resource types called Registration Definitions (RDs). We allow our users to retrieve RDs for a particular scope or for an entire tenant. This means our service allows for the following GET RDs URIs:

- /{scope}/providers/Microsoft.ManagedServices/registrationDefinitions
- /providers/Microsoft.ManagedServices/registrationDefinitions

Since according to the Swagger docs, we can't have an {optionalScope} parameter (i.e. we cannot have a parameter in the path that is required=false), what is the next best course? Do we specify the above two as two separate Swagger paths (even though they return the same type of result)? Is there a better option?

**Review by JeffreyRichter — APPROVED** [review]

**Comment by ArcturusZhang** [comment]
> Hi @ArcturusZhang , I wanted to ask this question directly to you as in case I don't get a response from the Swagger tools team alias. Here it is:
> 
> I tried looking for this in the docs but couldn't find a similar scenario. For our service we have these Azure resource types called Registration Definitions (RDs). We allow our users to retrieve RDs for a particular scope or for an entire tenant. This means our service allows for the following GET RDs URIs:
> 
> * /{scope}/providers/Microsoft.ManagedServices/registrationDefinitions
> * /providers/Microsoft.ManagedServices/registrationDefinitions
> 
> Since according to the Swagger docs, we can't have an {optionalScope} parameter (i.e. we cannot have a parameter in the path that is required=false), what is the next best course? Do we specify the above two as two separate Swagger paths (even though they return the same type of result)? Is there a better option?

taking the existing examples, we need to have both of these paths in swagger. We have some similar cases in the [policy swaggers](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/resources/resource-manager/Microsoft.Authorization/stable/2020-09-01/policyDefinitions.json). It is not exactly the same but quite similar

**Comment by skayani** [comment]
@ArcturusZhang Hey, I've gone ahead and added two paths for our marketplaceRegistrationDefinitions endpoints. I initially received breaking changes approval for removing scope. Will the breaking changes approval still apply to this PR?

**Review by ArcturusZhang — APPROVED** [review]

**Pull request merged**